openfecli/commands/gather.py

```diff
@@ -0,0 +1,133 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/openfe
+
+import click
+from openfecli import OFECommandPlugin
+import pathlib
+from openfecli.utils import write
+
+
+def is_results_json(f):
+    # sanity check on files before we try and deserialize
+    return 'estimate' in open(f, 'r').read(20)
+
+
+def load_results(f):
+    # path to deserialized results
+    import json
+    from gufe.tokenization import JSON_HANDLER
+
+    return json.load(open(f, 'r'), cls=JSON_HANDLER.decoder)
+
+
+def get_names(result):
+    # Result to list of ligand names
+    return tuple(list(result['unit_results'].values())[0]['name'].split()[:2])
+
+
+def get_type(f):
+    if 'solvent' in f:
+        return 'solvent'
+    elif 'vacuum' in f:
+        return 'vacuum'
+    else:
+        return 'complex'
+
+@click.command(
+    'gather',
+    short_help="Gather DAG result jsons for network of RFE results into single TSV file"
+)
+@click.argument('rootdir',
+                type=click.Path(dir_okay=True, file_okay=False,
+                                         path_type=pathlib.Path),
+                required=True)
+@click.argument('output',
+                type=click.File(mode='w'),
+                required=True)
+def gather(rootdir, output):
+    """Gather DAG result jsons of relative calculations and write to single tsv file
+
+    Will walk ROOTDIR recursively and find all results files ending in .json (i.e those produced by the quickrun
+    command).  Each of these is a single leg (e.g. L1-L2 in solvent) of a free energy calculation.
+
+    Will produce a **tab** separated file with 3 columns;
+    - a description of the measurement (e.g. DDGhyd(A, B) or DGsol(A, B))
+    - the estimated value (in kcal/mol)
+    - the uncertainty on the value (also kcal/mol)
+
+    Paired legs of simulations will be combined to give the DDG values between two ligands in the corresponding phase,
+    producing either binding ('DDGbind') or hydration ('DDGhyd') relative free energies.  These will be reported as
+    'DDGbind(B,A)' meaning DGbind(B) - DGbind(A), the difference in free energy of binding for ligand B compared to
+    ligand A.
+
+    Individual leg results will be also be written.  These are reported as either DGvacuum(A,B) DGsolvent(A,B) or
+    DGcomplex(A,B) for the vacuum, solvent or complex free energy of transmuting ligand A to ligand B.
+
+    Use output = '-' to stream to stdout.
```

Comment:
@IAlibay @RiesBen can you both read this and see if it would make sense to you both?